### PR TITLE
dotenv update

### PR DESCRIPTION
**Pull request by rasmusebbesen**
- Fixed issue on mailgun init
- Added dotenv extension to Environment from vapor-community/vapor-ext

### CODE REVIEW

File: Sources/App/Utilities/Environment+dotenv.swift [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                continue
+            }
+
+            // ignore lines that appear empty
+            if line.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
+                continue
+            }
+
+            // extract key and value which are separated by an equals sign
+            let parts = line.components(separatedBy: "=")
```
Comment (siemensikkema) [inline Sources/App/Utilities/Environment+dotenv.swift]: As you said, we know that this causes problems with values containing `=` signs. Here's how we could fix that:

```swift
guard
    let indexOfKeyValueSeparator = line.firstIndex(of: "="), // has to contain at least one "="
    indexOfKeyValueSeparator != line.startIndex // line must not start with "=" (ie. key must not be empty)
else {
    continue
}

let key = String(line.prefix(upTo: indexOfKeyValueSeparator))
var value = String(line.suffix(from: line.index(after: indexOfKeyValueSeparator)))
```

File: Sources/App/Utilities/Environment+dotenv.swift [hunk 2]
```diff
@@ -0,0 +1,77 @@
+//
+// Copy of extension from vapor-community/vapor-ext
```
Comment (siemensikkema) [inline Sources/App/Utilities/Environment+dotenv.swift]: If we're going with an adapted version maybe this should read "Adapted from vapor-community/vapor-ext" or similar

**Review by siemensikkema — COMMENTED** [review]
👍 on the Mailgun fix. I think we should get my suggestion on improving the dotenv support in. What do you think?

File: Sources/App/Utilities/Environment+dotenv.swift [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                value.remove(at: value.startIndex)
+                value.remove(at: value.index(before: value.endIndex))
+                value = value.replacingOccurrences(of: "\\\"", with: "\"")
+            }
+            
+            // remove surrounding single quotes from value & convert remove escape character before any embedded quotes
+            if value[value.startIndex] == "'" && value[value.index(before: value.endIndex)] == "'" {
+                value.remove(at: value.startIndex)
+                value.remove(at: value.index(before: value.endIndex))
+                value = value.replacingOccurrences(of: "'", with: "'")
```
Comment (siemensikkema) [inline Sources/App/Utilities/Environment+dotenv.swift]:
```suggestion
                value = value.replacingOccurrences(of: "\\'", with: "'")
```

**Review by siemensikkema — COMMENTED** [review]
Shouldn't we also add this to the start of the configure function?

```swift
    // Load variables from .env file
    Environment.dotenv()
```

**Comment by siemensikkema** [comment]
@rasmusebbesen Try running `swift package update` and committing Package.resolved to fix CI

File: Sources/App/Utilities/Environment+dotenv.swift (hunk 1 above)
Comment (rasmusebbesen) [reply Sources/App/Utilities/Environment+dotenv.swift]: We must return in a guard's else block, right?

Comment (siemensikkema) [reply Sources/App/Utilities/Environment+dotenv.swift]: You must exit the current scope. That can be `return` (in a function), `continue` (in a loop), `throw ...`, or `fatalError()` and maybe others I can't think of right now

Comment (rasmusebbesen) [reply Sources/App/Utilities/Environment+dotenv.swift]: Alright updated this

File: Sources/App/Utilities/Environment+dotenv.swift (hunk 2 above)
Comment (rasmusebbesen) [reply Sources/App/Utilities/Environment+dotenv.swift]: updated this

File: Sources/App/Utilities/Environment+dotenv.swift (hunk 3 above)
Comment (rasmusebbesen) [reply Sources/App/Utilities/Environment+dotenv.swift]: updated this

**Comment by rasmusebbesen** [comment]
Added all of the above :)

**Review by siemensikkema — DISMISSED** [review]
Looks good! Let's get it in :shipit:

**Comment by siemensikkema** [comment]
@rasmusebbesen aargh, CI fails. I believe we need 2 different cache keys for linux and mac. We've been doing this in a few places and it seems to solve these (apparently) random issues. Use `macos-v1-spm-deps-{{ checksum "Package.resolved" }}` and `linux-v1-spm-deps-{{ checksum "Package.resolved" }}`. Remember to change both occurrences of each cache key :)

**Review by siemensikkema — APPROVED** [review]

**Pull request merged**